src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/PackageReferenceProject.cs

```diff
@@ -162,7 +162,9 @@ public async Task<ProjectPackages> GetInstalledAndTransitivePackagesAsync(bool i
             {
                 // Compute Transitive Origins
                 Dictionary<string, TransitiveEntry> transitiveOrigins;
-                if (IsInstalledAndTransitiveComputationNeeded || TransitiveOriginsCache == null)
+                if (IsInstalledAndTransitiveComputationNeeded // Cache invalidation
+                    || TransitiveOriginsCache == null // If any data race left the cache as null
+                    || (!TransitiveOriginsCache.Any() && calculatedTransitivePackages.Any() && includeTransitiveOrigins)) // We have transitive packages, but no transitive origins and the call is requesting transitive origins
```

Comment:
It looks like you left 1 more my comment: 
It looks we're already checking includeTransitiveOrigins is true condition at line 161, so not sure if we need to check again.

Reply:
Oh right! Thank you for pointing out this!